src/main/perl/FileWriter.pm

```diff
@@ -210,15 +246,33 @@ sub close
                 $self->report ($diff);
             }
 
-            if (*$self->{options}->{noaction}) {
-                *$self->{options}->{contents} = $$content_ref;
+            if ($options->{noaction}) {
+                $msg = 'would have been';
+                $self->debug(1, "File $filename with NoAction=1");
+            } else {
+                my $opts = {
+                    file => $filename,
+                    input => $content_ref,
+                };
+                # group is handled separately
+                foreach my $name (qw(mode mtime backup owner)) {
+                    $opts->{$name} = $options->{$name} if exists($options->{$name});
+                };
+                # No need to check if owner exists. :groupname is supported
+                $opts->{owner} .= ":$options->{group}" if exists($options->{group});
+
+                eval {
+                    File::AtomicWrite->write_file($opts);
+                    $modified = 1;
+                };
+                if ($@) {
+                    $self->verbose("AtomicWrite gave error: $@");
```

Comment:
Is this only verbose because throw_error takes care of generating the actual error? In which case debug would be more appropriate to avoid the duplicate message?